pysignalscope/handleimpedance.py

```diff
@@ -0,0 +1,460 @@
+
+        ax2.grid()
+        ax2.set(xlabel='Frequency in Hz', ylabel='Phase in °')
+        ax2.legend()
+        plt.tight_layout()
+        plt.show()
+
+    @staticmethod
+    def plot_inductance_and_ac_resistance(channel_list: List) -> None:
+        """
+        Plot and compare inductance (in uH) and ac_resistance (Ohm) of impedance channels.
+
+        :param channel_list: List with impedances
+        :type channel_list: List
+        :return: None
+        :rtype: None
+        """
+        fig, (ax1, ax2) = plt.subplots(nrows=2, ncols=1, sharex=True)
+        for channel in channel_list:
+            ax1.semilogx(channel.channel_frequency,
+                         1e6 * channel.channel_impedance * np.sin(np.deg2rad(channel.channel_phase)) / channel.channel_frequency / 2 / np.pi,
+                         label=channel.channel_label, color=channel.channel_color, linestyle=channel.channel_linestyle)
+            ax2.semilogx(channel.channel_frequency,
+                         channel.channel_impedance * np.cos(np.deg2rad(channel.channel_phase)),
+                         label=channel.channel_label, color=channel.channel_color, linestyle=channel.channel_linestyle)
+
+        ax1.grid()
+        ax1.legend()
+        ax1.set(xlabel='Frequency in Hz', ylabel=r'Inductance in $\rm{\mu}$H')
+
+        ax2.grid()
+        ax2.set(xlabel='Frequency in Hz', ylabel=r'AC Resistance in $\Omega$')
+        ax2.legend()
+        plt.show()
+
+    @staticmethod
+    def calc_re_im_parts(channel: Impedance, show_figure: bool = True):
+        """
+        Calculate real and imaginary part of Impedance measurement.
+
+        :param channel: Impedance object
+        :type channel: Impedance
+        :param show_figure: Plot figure if true
+        :type show_figure: bool
+        :return: List with [(channel_frequency, frequency_real_part), (channel_frequency, frequency_imag_part)]
+        :rtype: List
+        """
+        frequency_real_part = []
+        frequency_imag_part = []
+        complex_impedance = []
+
+        for count_frequency, _ in enumerate(channel.channel_frequency):
+            impedance = channel.channel_impedance[count_frequency] * np.exp(1j * channel.channel_phase[count_frequency] * 2 * np.pi / 360)
+            complex_impedance.append(impedance)
+            frequency_real_part.append(np.real(complex_impedance[count_frequency]))
+            frequency_imag_part.append(np.imag(complex_impedance[count_frequency]))
+
+        if show_figure:
+            fig, (ax1, ax2) = plt.subplots(nrows=2, ncols=1, sharex=True)
+            ax1.loglog(channel.channel_frequency, frequency_real_part)
+            ax1.grid()
+            ax1.set_xlabel('frequency in Hz')
+            ax1.set_ylabel('real (impedance)')
+
+            ax2.loglog(channel.channel_frequency, frequency_imag_part)
+            ax2.set_xlabel('frequency in Hz')
+            ax2.set_ylabel('imag (impedance)')
+            ax2.grid()
+            fig.show()
+
```

Comment:
For me, it is unclear, if the return of variable channel returns the content of the variable (then it is ok) or only the reference of the variable (then it is nok)

Reply:
deepcopy is added